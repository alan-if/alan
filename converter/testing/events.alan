SYNTAX take = 'take' (o)*.
VERB take
    DOES
	LOCATE o IN inventory.
	"Taken."
END VERB.


SYNTAX drop = 'drop' (o)*.
VERB drop
    CHECK OBJECT IN inventory
	ELSE "You don't have the $o."
    DOES
	LOCATE OBJECT HERE.
END VERB.


SYNTAX l = 'l'.
VERB l
    DOES
	LOOK.
END VERB.


SYNTAX i = 'i'.
VERB i
  DOES
    LIST inventory.
END VERB i.


SYNTAX q = 'q'.
VERB q
  DOES
    QUIT.
END VERB q.


SYNTAX z = 'z'.
VERB z
  DOES
    ""
END VERB z.

EVENT e1
  "Happens every turn."
  SCHEDULE e1 AFTER 1.
END EVENT e1.

EVENT e2
  "Happens every 2."
  SCHEDULE e2 AFTER 2.
END EVENT e2.

LOCATION loc
END LOCATION loc.

START AT loc.
  SCHEDULE e1 AFTER 1.
  SCHEDULE e2 AFTER 2.
