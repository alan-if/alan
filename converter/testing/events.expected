########## events ##########

Syntax
  take = 'take' (o)*.

Verb take
  Does
    Locate o In inventory.
    "Taken."
End Verb.

Syntax
  drop = 'drop' (o)*.

Verb drop
  Check object In inventory
    Else 
      "You don't have the $o."
  Does
    Locate object Here.
End Verb.

Syntax
  l = 'l'.

Verb l
  Does
    Look.
End Verb.

Syntax
  i = 'i'.

Verb i
  Does
    List inventory.
End Verb.

Syntax
  q = 'q'.

Verb q
  Does
    Quit.
End Verb.

Syntax
  z = 'z'.

Verb z
  Does
    ""
End Verb.

Event e1
  "Happens every turn."
  Schedule e1 After 1.
End Event e1.

Event e2
  "Happens every 2."
  Schedule e2 After 2.
End Event e2.


The loc Isa location
  
End The loc.

Start At loc.
  Schedule e1 After 1.
  Schedule e2 After 2.
  
